packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -1359,6 +1365,10 @@ class WebpackCLI {
             process.exit(2);
         }
 
+        if (typeof options.nodeEnv === 'string') {
```

Comment:
Let' create new method `applyNodeEnv` and run it here https://github.com/webpack/webpack-cli/blob/bcf1c2c0f76642d9bfc63a78f3bac3cd324b5547/packages/webpack-cli/lib/webpack-cli.js#L1656 to avoid mixing logic